plugins/sfThumbnailPlugin/lib/sfImageMagickAdapter.class.php

```diff
@@ -217,6 +217,25 @@ public function loadFile($thumbnail, $image)
     return true;
   }
 
+  /**
+   * Determine file mime type using the PHP fileinfo library.
+   *
+   * @param string  file we want the mime type for
+   *
+   * @return string  Mime type
+   */
+  public static function getMimeType($file)
+  {
+    // Use fileinfo to figure out file mimetype.
+    $file_info = new finfo(FILEINFO_MIME);
+    $mimeInfo = $file_info->buffer(file_get_contents($file));
```

Comment:
Good catch.  I have updated and tested with big files and there is a noticeable performance increase.